File: BridgeToTurbin3/airdrop/transfer.ts
import { Transaction, SystemProgram, Connection, Keypair, LAMPORTS_PER_SOL, sendAndConfirmTransaction, PublicKey } from "@solana/web3.js"

import wallet from "./dev-wallet.json"

// Import dev wallet which has SOL in it
const from = Keypair.fromSecretKey(new Uint8Array(wallet));

// Define Second Account Public Key
/* Make sure you are in the right directory in the terminal before running "yarn keygen" to create a second wallet to receive a tx */
const to = new PublicKey("4aA18qjgPisC99WzVExCxaBqE2o9uvPc34N78CaK3qW3");


//Create a Solana devnet connection
const connection = new Connection("https://api.devnet.solana.com");


(async () => {
    try {
        // create a transaction and add instruction
        const transaction = new Transaction().add(
            SystemProgram.transfer({
                fromPubkey: from.publicKey,
                toPubkey:  to,
                lamports: LAMPORTS_PER_SOL/100,
            })
        );
        transaction.recentBlockhash = (await connection.getLatestBlockhash('confirmed')).blockhash;
        transaction.feePayer = from.publicKey;
        
        // Sign transaction, broadcast, and confirm
        const signature = await sendAndConfirmTransaction(
            connection,
            transaction,
            [from]
        );
        console.log(`Success! Check out your TX here: 
        https://explorer.solana.com/tx/${signature}?cluster=devnet`);
    } catch(e) {
        console.error(`Oops, something went wrong: ${e}`)
    }
})();

// terminal -> yarn transfer

File: RustRegistration/src/transfer.rs
use solana_client::rpc_client::RpcClient;
use solana_program::{pubkey::Pubkey, system_instruction::transfer};
use solana_sdk::{signature::{read_keypair_file, Signer},transaction::Transaction,};
use std::str::FromStr;

const RPC_URL: &str = "https://api.devnet.solana.com";

pub fn transfer_sol() {
    // Import our keypair
    let keypair = read_keypair_file("dev-wallet.json").expect("Couldn't find wallet file");
    // let _check_public_key = keypair.pubkey().to_string();
    
    // Define our WBA public key (the wallet which will receive tokens):
    let to_pubkey = Pubkey::from_str("Gy7nU5LzZkmP2yxVjDEYw3x7u17geBJSgSLiGsdXFMxw").unwrap();

    // Create a Solana devnet connection
    let rpc_client = RpcClient::new(RPC_URL);

    // Get recent blockhash
    let recent_blockhash = rpc_client
        .get_latest_blockhash()
        .expect("Failed to get recent blockhash");

    // create a transaction and sign it
    let transaction = Transaction::new_signed_with_payer(
        &[transfer(&keypair.pubkey(), &to_pubkey, 100_000_000)],
        Some(&keypair.pubkey()),
        &vec![&keypair],
        recent_blockhash,
    );

    // Send the transaction
    let signature = rpc_client
        .send_and_confirm_transaction(&transaction)
        .expect("Failed to send transaction");

    println!(
        "Success! Check out your TX here:
    https://explorer.solana.com/tx/{}/?cluster=devnet",
        signature
    );
}

// region :  Note to Self
/*
    Why do we need recent blockhash? In order to sign a transaction we need recent blockhash as well as keypair.
 */
//endregion :  Note to Self
